### PR TITLE
Jason dev4

### PR DESCRIPTION
**Pull request by jason-alouda**
Get post requests working

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java [hunk 1]
```diff
@@ -20,40 +20,57 @@
...
+import com.google.gson.Gson;
 
-/** Servlet responsible for listing tasks. */
-@WebServlet("/list-tasks")
-public class ListTasksServlet extends HttpServlet {
 
+
+@WebServlet("/comments")
+public class CommentsServlet extends HttpServlet {
+  private ArrayList<String> comments;
```
[the remark this answers is not in the document]
Comment (jason-alouda) [reply portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java]: removed

File: portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java [hunk 2]
```diff
@@ -20,40 +20,57 @@
...
+    commentEntity.setProperty("timestamp", timestamp);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(commentEntity);
 
+    // json format file conversion
+    Gson gson = new Gson();
+    String json = gson.toJson(comments);
     response.setContentType("application/json;");
-    response.getWriter().println(gson.toJson(tasks));
+    response.getWriter().println(json);
```
[the remark this answers is not in the document]
Comment (jason-alouda) [reply portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java]: removed

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -34,3 +34,31 @@ function addRandomQuote() {
...
+
+  //const response = await fetch('/data');
+  //const greeting = await response.text();
+  //document.getElementById('greeting-container').innerHTML = greeting;
+
+  fetch('/data').then(response => response.json()).then((greetings) => {
+    const greetingElem = document.getElementById('greeting-container');
+    var index = Math.floor(Math.random() * 10) % greetings.length;
+    greetingElem.innerHTML = greetings[index];
+  });
```
[the remark this answers is not in the document]
Comment (jason-alouda) [reply portfolio/src/main/webapp/script.js]: resolved

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -34,3 +34,31 @@ function addRandomQuote() {
...
+
+  //const response = await fetch('/data');
+  //const greeting = await response.text();
+  //document.getElementById('greeting-container').innerHTML = greeting;
+
+  fetch('/view-comments').then(response => response.json()).then((comments) => {
+    const commentElem = document.getElementById('comment-container');
+    var index = Math.floor(Math.random() * 10) % comments.length;
+    commentElem.innerText = comments[index].user_comment;
+  });
```
[the remark this answers is not in the document]
Comment (jason-alouda) [reply portfolio/src/main/webapp/script.js]: Display all comments - resolved!

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -34,3 +34,34 @@ function addRandomQuote() {
   const quoteContainer = document.getElementById('quote-container');
   quoteContainer.innerText = quote;
 }
+
+// Fetch greetings from server and add a random one to the page.
+async function getGreeting() {
+
+  //const response = await fetch('/data');
+  //const greeting = await response.text();
+  //document.getElementById('greeting-container').innerHTML = greeting;
```
Comment (classm-sps) [inline portfolio/src/main/webapp/script.js]: nit: delete commented code

File: portfolio/src/main/webapp/script.js [hunk 6]
```diff
@@ -34,3 +34,34 @@ function addRandomQuote() {
...
+    greetingElem.innerHTML = greetings[index];
+  });
+}
+
+// Fetch comments from server and add a random one to page.
+async function getComments() {
+
+  //const response = await fetch('/data');
+  //const greeting = await response.text();
+  //document.getElementById('greeting-container').innerHTML = greeting;
```
Comment (classm-sps) [inline portfolio/src/main/webapp/script.js]: nit: delete commented code

**Pull request merged**